src/UI/StaffCraftingMenu.cpp

```diff
@@ -902,12 +902,34 @@ namespace UI
 		}
 
 		if (selected.morpholith->data->GetObject()->IsSoulGem()) {
-			player->RemoveItem(
-				selected.morpholith->data->GetObject(),
-				1,
-				RE::ITEM_REMOVE_REASON::kRemove,
-				nullptr,
-				nullptr);
+			const auto
+				baseMorpholith = selected.morpholith->data->GetObject()->As<RE::BGSKeywordForm>();
```

Comment:
`As` is redundant here, we have already checked that it is a soul gem so we can use `static_cast<RE::TESSoulGem*>(...)` instead.